modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/validation/ParameterModelValidatorTestCase.java

```diff
@@ -140,6 +141,21 @@ public void invalidConfigOverrideWithDefault() {
     validate(extensionModel, validator);
   }
 
+
+  @Test
+  public void invalidEnumDefaultValue() {
+    expectedException.expect(IllegalModelDefinitionException.class);
+    expectedException
+        .expectMessage("Parameter 'url' in the operation 'dummyOperation' has a default value which is not listed as an available option");
+
+    when(invalidParameterModel.isRequired()).thenReturn(false);
+    when(invalidParameterModel.getDefaultValue()).thenReturn("default");
+    when(invalidParameterModel.getType()).thenReturn(toMetadataType(HealthStatus.class));
+    mockParameters(operationModel, invalidParameterModel);
+
+    validate(extensionModel, validator);
+  }
+
```

Comment:
Add a test where the default value is an expression and a test where the default value is a valid enum value.